iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -24,20 +24,29 @@
 /// @brief publisher handle
 typedef struct cpp2c_Publisher* iox_pub_t;
 
+/// @brief options to be set for a publisher
+typedef struct
+{
+    // size of the history chunk queue
+    uint64_t historyCapacity;
+
+    // name of the node the publisher belongs to
+    const char* nodeName;
+} iox_publisher_options_t;
+
 /// @brief creates a publisher handle
 /// @param[in] self pointer to preallocated memory of size = sizeof(iox_pub_storage_t)
 /// @param[in] service serviceString
 /// @param[in] instance instanceString
 /// @param[in] event eventString
-/// @param[in] historyCapacity size of the history chunk queue
-/// @param[in] nodeName name of the node the publisher belongs to
+/// @param[in] options publisher options set by the user
 /// @return handle of the publisher
+/// @attention the options must be set to valid values
```

Comment:
Can do, but it is easier to forget this in case of structs than an int variable where you a least have to pass something which you hopefully initialized... It is easy to forget to initialize a struct member in C.